### PR TITLE
DD-315 xml formatted provenance

### PR DESCRIPTION
**Pull request by jo-pol**
Fixes DD-315 xml formatted provenance

When applied it will
--------------------
* 
* 
* 

Where should the reviewer @DANS-KNAW/easy start?
------------------------------------------------

How should this be manually tested?
-----------------------------------

Related pull requests on github
-------------------------------

repo                       | PR                | note
-------------------------- | ----------------- | ----
easy-                      | [PR#](PRlink)     | ..

### CODE REVIEW

File: src/main/scala/nl.knaw.dans.easy.bag2deposit/EasyConvertBagToDepositApp.scala [hunk 1]
```diff
@@ -82,7 +82,7 @@ class EasyConvertBagToDepositApp(configuration: Configuration) extends DebugEnha
       datasetId = props.getString("identifier.fedora", "")
       ddmOut <- configuration.ddmTransformer.transform(ddmIn, datasetId)
       _ = Provenance(ddmIn, ddmOut, s"${ getClass.getSimpleName } ${ configuration.version }")
-        .foreach(s => logger.info(s))
+        .foreach(xml => (bagDir / "metadata" / "prov.xml").writeText(xml.serialize))
```
Comment (jo-pol) [inline src/main/scala/nl.knaw.dans.easy.bag2deposit/EasyConvertBagToDepositApp.scala:85]: not touched by unit tests

File: src/main/scala/nl.knaw.dans.easy.bag2deposit/EasyConvertBagToDepositApp.scala [hunk 2]
```diff
@@ -56,7 +56,7 @@ class EasyConvertBagToDepositApp(configuration: Configuration) extends DebugEnha
   def logMatchedReports(): Unit = {
     val uuidToReportLabel = configuration.ddmTransformer.reportRewriteRule.reportMap
       .map(r => r.uuid -> r.label).toMap
-    reportMatches.foreach { case (reportUuid, matches) =>
+    reportMatches.withFilter(_._2.isEmpty).foreach { case (reportUuid, matches) =>
```
Comment (jo-pol) [inline src/main/scala/nl.knaw.dans.easy.bag2deposit/EasyConvertBagToDepositApp.scala:59]: this reduces the check list at the end of the log

File: src/test/scala/nl/knaw/dans/easy/bag2deposit/ddm/TitlesSpec.scala [hunk 3]
```diff
@@ -50,44 +51,60 @@ class TitlesSpec extends AnyFlatSpec with FileSystemSupport {
...
-
     (testDir / "missed.txt").write(
-      titlesPerDataset.map { case (id, t) =>
-        t.filterNot(title => rule.reportMap.exists(m => title.toLowerCase.matches(".*" + m.regexp)))
-          .filter(_.toLowerCase.matches(s".*$keyword[^0-9]*${ nrRegexp }(:.*)?"))
-          .map(title => s"\n$id\t$title")
-          .mkString("")
-      }.mkString(""))
+      otherMissed(true).mkString("\n")
+    )
```
Comment (jo-pol) [inline src/test/scala/nl/knaw/dans/easy/bag2deposit/ddm/TitlesSpec.scala:107]: These changes reduce execution time by 50%. The test is ignored by default.

File: src/main/scala/nl.knaw.dans.easy.bag2deposit/EasyConvertBagToDepositApp.scala (hunk 1 above)
Comment (jo-pol) [reply src/main/scala/nl.knaw.dans.easy.bag2deposit/EasyConvertBagToDepositApp.scala:85]: We might revert this line so merge does not have to wait for updates of schemas and bag validation.

File: src/test/scala/nl/knaw/dans/easy/bag2deposit/ddm/ProvenanceSpec.scala [hunk 4]
```diff
@@ -82,35 +84,25 @@ class ProvenanceSpec extends AnyFlatSpec with Matchers {
...
+            <ddm:subject xml:lang="nl" valueURI="https://data.cultureelerfgoed.nl/term/id/abr/f182d72c-2d22-47ae-b799-26dea01e770c" subjectScheme="Archeologisch Basis Register" schemeURI="https://data.cultureelerfgoed.nl/term/id/abr/b6df7840-67bf-48bd-aa56-7ee39435d2ed">akker / tuin</ddm:subject>
```
Comment (lindareijnhoudt) [inline src/test/scala/nl/knaw/dans/easy/bag2deposit/ddm/ProvenanceSpec.scala:101]: in PR#16 i've changed the subjectScheme 
```suggestion
            <ddm:subject xml:lang="nl" valueURI="https://data.cultureelerfgoed.nl/term/id/abr/f182d72c-2d22-47ae-b799-26dea01e770c" subjectScheme="ABR Complextypen" schemeURI="https://data.cultureelerfgoed.nl/term/id/abr/b6df7840-67bf-48bd-aa56-7ee39435d2ed">akker / tuin</ddm:subject>
```

File: src/test/scala/nl/knaw/dans/easy/bag2deposit/ddm/ProvenanceSpec.scala [hunk 5]
```diff
@@ -82,35 +84,25 @@ class ProvenanceSpec extends AnyFlatSpec with Matchers {
...
...ddm:reportNumber schemeURI="https://data.cultureelerfgoed.nl/term/id/abr/7a99aaba-c1e7-49a4-9dd8-d295dbcc870e" valueURI="https://data.cultureelerfgoed.nl/term/id/abr/fcff6035-9e90-450f-8b39-cf33447e6e9f" subjectScheme="ABR Rapporten" reportNo="456">Rapport 456</ddm:reportNumber><ddm:reportNumber schemeURI="https://data.cultureelerfgoed.nl/term/id/abr/7a99aaba-c1e7-49a4-9dd8-d295dbcc870e" valueURI="https://data.cultureelerfgoed.nl/term/id/abr/90f3092a-818e-4db2-8467-35b64262c5b3" subjectScheme="ABR Rapporten" reportNo="2859">Transect-rapport 2859</ddm:reportNumber><ddm:temporal xml:lang="nl" valueURI="https://data.cultureelerfgoed.nl/term/id/abr/330e7fe0-a1f7-43de-b448-d477898f6648" subjectScheme="Archeologisch Basis Register" schemeURI="https://data.cultureelerfgoed.nl/term/id/abr/b6df7840-67bf-48bd-aa56-7ee39435d2ed">Vroege Middeleeuwen A</ddm:temporal><ddm:subject xml:lang="nl" valueURI="https://data.cultureelerfgoed.nl/term/id/abr/6ae3ab19-49ca-44a7-8b65-3a3395014bb9" subjectScheme="Archeologisch Basis Register" schemeURI="https://data.cultureelerfgoed.nl/term/id/abr/b6df7840-67bf-48bd-aa56-7ee39435d2ed">veenwinning (inclusief zouthoudend veen t.b.v. zoutproductie)</ddm:subject>
```
Comment (lindareijnhoudt) [inline src/test/scala/nl/knaw/dans/easy/bag2deposit/ddm/ProvenanceSpec.scala:100]: in PR#16 i've changed the subjectScheme
```suggestion
            <ddm:reportNumber schemeURI="https://data.cultureelerfgoed.nl/term/id/abr/7a99aaba-c1e7-49a4-9dd8-d295dbcc870e" valueURI="https://data.cultureelerfgoed.nl/term/id/abr/fcff6035-9e90-450f-8b39-cf33447e6e9f" subjectScheme="ABR Rapporten" reportNo="456">Rapport 456</ddm:reportNumber><ddm:reportNumber schemeURI="https://data.cultureelerfgoed.nl/term/id/abr/7a99aaba-c1e7-49a4-9dd8-d295dbcc870e" valueURI="https://data.cultureelerfgoed.nl/term/id/abr/90f3092a-818e-4db2-8467-35b64262c5b3" subjectScheme="ABR Rapporten" reportNo="2859">Transect-rapport 2859</ddm:reportNumber><ddm:temporal xml:lang="nl" valueURI="https://data.cultureelerfgoed.nl/term/id/abr/330e7fe0-a1f7-43de-b448-d477898f6648" subjectScheme="ABR Perioden" schemeURI="https://data.cultureelerfgoed.nl/term/id/abr/b6df7840-67bf-48bd-aa56-7ee39435d2ed">Vroege Middeleeuwen A</ddm:temporal><ddm:subject xml:lang="nl" valueURI="https://data.cultureelerfgoed.nl/term/id/abr/6ae3ab19-49ca-44a7-8b65-3a3395014bb9" subjectScheme="ABR Complextypen" schemeURI="https://data.cultureelerfgoed.nl/term/id/abr/b6df7840-67bf-48bd-aa56-7ee39435d2ed">veenwinning (inclusief zouthoudend veen t.b.v. zoutproductie)</ddm:subject>
```

**Pull request merged**